lib/transaction/transaction.js

```diff
@@ -828,17 +828,22 @@ Transaction.prototype._updateChangeOutput = function () {
   if (!_.isUndefined(this._changeIndex)) {
     this._removeOutput(this._changeIndex)
   }
+  this._changeIndex = this.outputs.length
+  this._addOutput(new Output({
+    script: this._changeScript,
+    satoshis: 0
+  }))
   var available = this._getUnspentValue()
   var fee = this.getFee()
   var changeAmount = available - fee
-  if (changeAmount > 0) {
+  this._removeOutput(this._changeIndex)
+  this._changeIndex = undefined
+  if (changeAmount >= Transaction.DUST_AMOUNT) {
```

Comment:
what are you changing here and why?

Reply:
Remove the dummy output created above. Now we check if there was any change left over. If there is enough change to cover the dust limit, we create the real change output.

Reply:
Neat, so doing `.change(adr)` will be ignored if the change value is below dust - right?

Reply:
@mathiasrw essentially yes. Next move is cleaning up datapay, it has some dangerous hacks to work around all this.